src/v/cluster/rm_stm.cc

```diff
@@ -373,6 +355,33 @@ ss::future<checked<model::term_id, tx_errc>> rm_stm::do_begin_tx(
         co_return tx_errc::fenced;
     }
 
+    std::optional<model::record_batch> batch;
+    if (is_transaction_ga()) {
+        batch = make_fence_batch(pid, tx_seq, transaction_timeout_ms);
+    } else {
+        batch = make_fence_batch(pid);
+    }
+
+    auto reader = model::make_memory_record_batch_reader(std::move(*batch));
+    auto r = co_await _c->replicate(
+      synced_term,
+      std::move(reader),
+      raft::replicate_options(raft::consistency_level::quorum_ack));
+
+    if (!r) {
+        vlog(
+          _ctx_log.error,
```

Comment:
should that be `warn` severity log entry ?

Reply:
good point, fixed